drivers/sensor/wsen_pdus_25131308XXXXX/wsen_pdus_25131308XXXXX.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_pdus_25131308xxxxx
+
+#include <stdlib.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+#include "wsen_pdus_25131308XXXXX.h"
+
+LOG_MODULE_REGISTER(WSEN_PDUS_25131308XXXXX, CONFIG_SENSOR_LOG_LEVEL);
+
+static int pdus_25131308XXXXX_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct pdus_25131308XXXXX_data *data = dev->data;
+	uint16_t pressure_dummy;
+
+	switch (chan) {
+	case SENSOR_CHAN_ALL: {
+		if (PDUS_getRawPressureAndTemperature(&data->sensor_interface, &data->pressure,
+						      &data->temperature) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		if (PDUS_getRawPressureAndTemperature(&data->sensor_interface, &pressure_dummy,
+						      &data->temperature) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	case SENSOR_CHAN_PRESS: {
+		if (PDUS_getRawPressure(&data->sensor_interface, &data->pressure) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	default:
+		LOG_ERR("Fetching is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int pdus_25131308XXXXX_channel_get(const struct device *dev, enum sensor_channel chan,
+					  struct sensor_value *value)
+{
+	struct pdus_25131308XXXXX_data *data = dev->data;
+	const struct pdus_25131308XXXXX_config *config = dev->config;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		int32_t temperatureMega = ((int32_t)(data->temperature - T_MIN_VAL_PDUS)) * 4272;
```

Comment:
Are the magic numbers here and throughout this function documented somewhere?

Reply:
I have added a comment that says that these value are mentioned in the user manual